packages/components/Callout/macos/GuardedEventMonitor.swift

```diff
@@ -0,0 +1,30 @@
+
+
+/// NSEvent localMonitors are an _old_ API that requires you to do some  memory management yourself.
```

Comment:
What's the gap in the NSEvent API? I think you are always fine if you just nil out your reference to the monitor after you remove it, right? At least that's what I've done elsewhere.

And this wrapper does provide some potential for confusion and misuse - `addLocalMonitorForEvents`  no-ops if there already is a monitor, even if you send in a different mask.

Reply:
Yep. Both of these are things I realized / people pointed out to me right after I merged. I didn't realize you could just nil out the reference and that would be sufficient. I rewrote the wrapper to do so in https://github.com/microsoft/fluentui-react-native/commit/b772dfe4542b7032f216bf5c2faa807c695fa2a7

There I added an extra comment that this internal class is only ever meant to be used for 1 monitor at a time, and I changed it so that we just nil out the monitor rather than the boolean check.


`addLocalMonitor` will never let you make more than 1 monitor per application? That's interesting...

Reply:
Sorry I meant that your wrapper function `addLocalMonitorForEvents` (not the system API) was the one that is confusing in that it would do nothing if you called it with a different mask because of the `if (!isMonitorAllocated)` check. Personally I would tend towards not having the wrapper. I think nil'ing out after removing is a pretty common pattern and it seems better to force people to remember that pattern than add our own abstraction on top that has other gotchas/patterns people would need to familiarize themselves with.